nix/tabry-hm-module.nix

```diff
@@ -0,0 +1,68 @@
+{ config, lib, pkgs, ... }:
+
+# This file contains a Home Manager module that installs tabry
+# and sets up tabry configuration files to be used by tabry
+#
+let
+
+  cfg = config.programs.tabry;
+
+  tabry = pkgs.callPackage ../default.nix {};
+
+  # converts /nix/store/.../foo.tabry to "foo"
+  commandNameFromTabryFilename = fileName: 
+    (builtins.replaceStrings [".tabry"] [""] (builtins.baseNameOf fileName));
+
+  mkInitFish = fileName: let
+    commandName = commandNameFromTabryFilename fileName;
+  in ''
+    tabry_completion_init ${commandName}
+  '';
+
+  tabryFileDir = pkgs.stdenv.mkDerivation {
+    name = "tabry-files";
+    dontUnpack = true;
+    phases = [ "installPhase" ];
+    installPhase = ''
+      mkdir -p $out
+      cp {${builtins.concatStringsSep "," cfg.tabryFiles}} $out
+    '';
+  };
+
+in {
+
+  options.programs.tabry = {
+    enable = lib.mkEnableOption "tabry, a tab completion library";
+    enableFishIntegration = lib.mkEnableOption "enables fish completions";
+    enableBashIntegration = lib.mkEnableOption "enables bash completions";
+    tabryFiles = lib.mkOption {
+      type = with lib.types; listOf path;
+      default = [];
+      description = ''
+        *.tabry files to be compiled to completion json
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    home.packages = [tabry];
+
+    # for each file, compile it to json
+    # then add the dir to $TABRY_IMPORTS_PATH
+
+    programs.fish.shellInit = lib.mkIf cfg.enableFishIntegration (
+      ''
+        set -x TABRY_IMPORTS_PATH "${tabryFileDir}:$TABRY_IMPORTS_PATH"
+        ${tabry}/bin/tabry fish | source
+        ${builtins.concatStringsSep "\n" (map mkInitFish cfg.tabryFiles)}
+      ''
+    );
+
+    programs.bash.initExtra = lib.mkIf cfg.enableBashIntegration (
+      ''
+        export TABRY_IMPORTS_PATH="${tabryFileDir}:$TABRY_IMPORTS_PATH"
```

Comment:
This should be TABRY_IMPORT_PATH (variable used in the rust version -- TABRY_IMPORTS_PATH is used in the ruby version), right?